### ISSUE TITLE
[ONNX] Clean up in-line imports in symbolic helper

### ISSUE THREAD

**Issue by justinchuby** [pytorch/pytorch#83981]
Clean up the rest of the in-line imports in `torch/onnx/symbolic_helpers.py`

Currently in `sybolic_helpers.py`, we have inline imports like these

https://github.com/pytorch/pytorch/blob/a4a55f5ea6403472a25b12a6e9b8f4f713e664a3/torch/onnx/symbolic_helper.py#L1246-L1260

where we import from specific opsets inline and use in particular functions. This goes against common best practices of importing on the top of the module. Instead, we should 

```python
from torch.onnx import symbolic_opset9 as opset9, symbolic_opset11 as opset11
```

and do

```python
def _index_fill_reshape_helper(g, self, dim, index):
    # 1. reshape index => [1, ..., 1, dim, 1, ..., 1]
    # 2. expand index => [..., dim, ...], same shape as self except for dim.
    # 3. expand value as well.
    # 4. apply onnx::scatter.
    if GLOBALS.export_onnx_opset_version <= 10:
        scatter = opset9.scatter
    else:
        scatter = opset11.scatter

    if self.type().dim() is None:
        return _unimplemented("index_fill", "input rank not accesible")
    self_dim = self.type().dim()
    dim_value = _parse_arg(dim, "i")
    unsqueezed_index = _unsqueeze_helper(
        g, index, [i for i in range(self_dim) if i != dim_value]
    )
    expanded_index_shape = scatter(
        g, g.op("Shape", self), 0, _unsqueeze_helper(g, dim, [0]), g.op("Shape", index)
    )
    expanded_index = opset9.expand(g, unsqueezed_index, expanded_index_shape, None)
    return expanded_index_shape, expanded_index
```

we should do this for all functions in the symbolic_helper module.

**Comment by 9gl** [comment]
Hi @justinchuby  ,
I'm interested in contributing to this issue, so before I start working
it, would you mind sparing your time explaining what the issue is about and pointing me to some resources to get started.

**Comment by justinchuby** [comment]
@9gl Thanks for offering to help! I updated the issue with more context. Please let me know if that's good enough to get you started.

**Comment by ShisuiUzumaki** [comment]
I would like to try my hand at this issue as well

**Comment by daspartho** [comment]
I would like to work on this issue. I noticed that this issue is still present. Can I contribute to this issue?

**Comment by justinchuby** [comment]
@daspartho thanks for offering to help! Looks like there is a circular import situation with the change thanks to @9gl 's investigation. We should pause and figure out a better way after refactoring.

**Comment by justinchuby** [comment]
Closed in favor of #85486
